### PR TITLE
feat: [IOPID-1539,IOPID-1784] - Integrate new DS in UpdateAppModal

### PR DESCRIPTION
**Pull request by ChrisMattew**
## Short description
Integrate new DS in UpdateAppModal and adjust key duplication in OperationResultScreenContent component

## List of changes proposed in this pull request
- Integrate new DS
- Adjust key duplication in OperationResultScreenContent component
- Added ad hoc Toast notification management in UpdateAppModal triggered by link opening error

## Screens
<details><summary>iOS simulator</summary>
<p>

 |  Open link success | Open link error    |
  | - | - | 
|<video src="https://github.com/pagopa/io-app/assets/94987118/b2df039a-4b3e-4fe0-8c81-8295f0c6ceda" /> | <video src="https://github.com/pagopa/io-app/assets/94987118/06f6cfc6-f99d-496f-b725-5fbef4a34b9d" /> |

</p>
</details>

<details><summary>Android device</summary>
<p>

 |  Open link success |  Open link error    |
  | - | - | 
|<video src="https://github.com/pagopa/io-app/assets/94987118/83b9cf75-8fb5-4ec9-aef2-d65cb16d392d" /> | <video src="https://github.com/pagopa/io-app/assets/94987118/21f04ebe-5910-43fb-8cb2-8eb257a1d280" /> |

</p>
</details>

## How to test

There are two ways is which test this case:
1. In `ts/screens/modal/RootModal.tsx` invert the `!props.isAppSupported` condition to display the UpdateAppModal screen
2. Testing in local environment - clone [io-dev-api-server](https://github.com/pagopa/io-dev-api-server) repo if you haven't did it yet and follow the [instructions](https://github.com/pagopa/io-app/blob/master/README.md#io-dev-api-server) - go in `assets/status/versionInfo.json` and change the version related to the OS of your device/simulator in order to display the UpdateAppModal in your app.
```json
"min_app_version": {
    "ios": "x.y.z",
    "android": "x.y.z"
 }
```

> [!IMPORTANT]  
> Testing with an android virtual device requires an additional setup before to start!

<details><summary>Android virtual device additional setup</summary>
<p>

In `ts/sagas/startup.ts` find and comment the code snippets shown here below:
```tsx
// this
const unsupportedDevice = yield* call(checkPublicKeyAndBlockIfNeeded);
if (unsupportedDevice) {
     return;
}

// and this
const isAssertionRefValid = yield* call(
    checkLollipopSessionAssertionAndInvalidateIfNeeded,
    publicKey,
    maybeSessionInformation
);
if (!isAssertionRefValid) {
     return;
}
```
then follow the process already explained in the section above.

</p>
</details>

### CODE REVIEW

**Comment by ChrisMattew** [comment]
@mariateresaventura  copy needs a review 🙏🏼

File: ts/components/screens/OperationResultScreenContent.tsx [hunk 1]
```diff
@@ -60,8 +60,8 @@ export const ComposedBodyFromArray = ({
   textAlign = "center"
 }: PropsComposedBody) => (
   <Body style={{ textAlign }}>
-    {subtitle.map(({ text, key, ...props }) => (
-      <Body key={key} {...props}>
+    {subtitle.map(({ text, key, ...props }, i) => (
+      <Body key={key ?? i} {...props}>
```
Comment (Vangaorth) [inline ts/components/screens/OperationResultScreenContent.tsx:64]: By implicitly using the index, we are not safe against another view in the same hierarchy that is constructed dynamically, using the index as key as well.

I suggest that we we append the index to a string, in order to reduce hitting duplicates (e.g., `` `CBFA_${i}` ``

**Review by Vangaorth — CHANGES_REQUESTED** [review]

File: ts/components/screens/OperationResultScreenContent.tsx (hunk 1 above)
Comment (ChrisMattew) [reply ts/components/screens/OperationResultScreenContent.tsx:64]: Hi @Vangaorth, thank you for this suggestion! I will integrate it as soon as possible!

**Comment by mariateresaventura** [comment]
> @mariateresaventura copy needs a review 🙏🏼

Done!

File: ts/screens/modal/UpdateAppModal.tsx [hunk 2]
```diff
@@ -3,160 +3,118 @@
...
   const openAppStore = useCallback(async () => {
     trackUpdateAppButton();
+
     try {
       await Linking.openURL(storeUrl);
     } catch (e) {
-      openWebUrl(webStoreURL, () => setError(true));
+      openWebUrl(webStoreURL, () => {
+        setIsError(true);
+      });
```
Comment (shadowsheep1) [inline ts/screens/modal/UpdateAppModal.tsx:68]:
```suggestion
      openWebUrl(webStoreURL, () => setIsError(true));
```

Comment (shadowsheep1) [reply ts/screens/modal/UpdateAppModal.tsx:68]: Just a suggestion ,)

**Review by Vangaorth — APPROVED** [review]

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**